scabha/cargo.py

```diff
@@ -126,9 +131,9 @@ class Parameter(object):
     aliases: Optional[List[str]] = ()
 
     # if true, create parent directories of file-type outputs if needed
-    mkdir: bool = True
-    # if true, create directory-type outputs themselves
-    mkdir_self: bool = False
+    mkdir_parent: bool = True
```

Comment:
My intuition is failing me on this change - can this break existing recipes? I almost wonder if this would be better as:
```python
mkdir: bool = True
mkdir_policy: str = "parents"   # With alternatives being "all", "child" etc. May also allow for more specific behaviour e.g. controlling exists_ok. 
```

Reply:
The policy suggestion is just that - whatever you feel is more appropriate. Are custom dataclasses supported at this level i.e. could we have a `MkdirPolicy` object which exposed some options? I agree that the default is going to cover most usage, but as you say, it may be nice to future-proof it we can. There will always be another weird use-case.

Reply:
Would propose a ``path_policies`` property which could also absorb ``access_parent_dir`` and ``write_parent_dir`` (only used in a couple of places in cult-cargo currently.)

Reply:
Sounds sensible to me if you are happy to do the implementing (I would offer but it would likely take me exponentially longer).